### PR TITLE
python312Packages.spylls: init at 0.1.5

### PR DESCRIPTION
**Pull request by ViZiD**
## Description of changes
Pure Python spell-checker, (almost) full port of Hunspell
https://github.com/zverok/spylls

https://github.com/zverok/spylls/blob/master/CHANGELOG.rst

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [x] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/development/python-modules/spylls/default.nix [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+
+  src = fetchPypi {
+    inherit pname version;
+    hash = "sha256-cEWJLcvTJNNoX2nFp2AGPnj7g5kTckzhgHfPCgyT8iA=";
+  };
+
+  postPatch = ''
+    substituteInPlace pyproject.toml \
+    --replace poetry.masonry.api poetry.core.masonry.api \
+    --replace "poetry>=" "poetry-core>="
```
Comment (natsukium) [inline pkgs/development/python-modules/spylls/default.nix:29]: Could you suggest the use of poetry-core to the upstream?
https://github.com/python-poetry/poetry-core?tab=readme-ov-file#why-is-this-required

Comment (ViZiD) [reply pkgs/development/python-modules/spylls/default.nix:29]: Done
https://github.com/zverok/spylls/pull/27

Comment (ViZiD) [reply pkgs/development/python-modules/spylls/default.nix]: @natsukium

File: pkgs/development/python-modules/spylls/default.nix [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+
+  nativeBuildInputs = [
+    pytest # hack for fix error: `pytest not installed`
+    pythonRelaxDepsHook
+  ];
+
+  pythonRelaxDeps = [
+    "pytest"
+  ];
+
```
Comment (natsukium) [inline pkgs/development/python-modules/spylls/default.nix:44]:
```suggestion
```
I can't reproduce such an error.

Comment (ViZiD) [reply pkgs/development/python-modules/spylls/default.nix:44]: @natsukium this error is only in version 0.1.5
Can you explain how pythonRelaxDeps work, if a package needs another package in version 0.2.0, but the latest version of the package is 0.2.5, do I need to commit version 0.2.0 and then 0.2.5?

**Comment by ViZiD** [comment]
@sikmir @Aleksanaa @natsukium

File: pkgs/development/python-modules/spylls/default.nix (hunk 2 above)
Comment (natsukium) [reply pkgs/development/python-modules/spylls/default.nix]: > if a package needs another package in version 0.2.0, but the latest version of the package is 0.2.5, do I need to commit version 0.2.0 and then 0.2.5?

No. `pythonRelaxDepsHook` simply ignores upstream version constraints.
It does not reference previous versions.

**Review by natsukium — APPROVED** [review]

**Pull request merged**